libs/temporal/temporal/timeline.h

```diff
@@ -40,8 +40,19 @@ class timecnt_t;
 
 void dump_stats (std::ostream&);
 
-/* 62 bit positional time value. Theoretically signed, but the intent is for to
- * always be positive. If the flag bit is set (i.e. ::flagged() is true), the
+/* A timepos_t designates an absolute position on the global timeline. It is
+ * measured since time zero, and it can thus only be positive. It is in one of
+ * two TimeDomains: AudioTime (wall time measured using superclock,
+ * proportional to sample count) or BeatTime (counting musical ticks, which are
+ * subdivided quarternotes or Beats, independent of tempo). Conversion between
+ * these time domains will use the global TempoMap.
+ *
+ * timepos_t is also sometimes used for storing time lengths not relative to time
+ * zero. In that case it is important to avoid using the TempoMap. Such
+ * durations can also be negative.
```

Comment:
Note: This "extended use" of timepos_t has confused me a lot. It seems like it would benefit a lot from  having a separate type for position-less deltas.

If my understanding is wrong, then we need a clarification even more.

Reply:
I think the final part of the comment is wrong at this point.

As your intuition suggests, it caused a LOT of problems and I believe we've eliminated all the places where we tried to do this. The comment can be edited.

Reply:
Ok, I will take a closer look at the current situation. Can you give a hint how such position-less durations are handled now? Do they just always have a position?

Meanwhile, it would be great if you can cherry-pick the parts you like.

Reply:
We now try to never use such position-less durations, but always use timecnt_t

Finding problem cases is non-trivial, AFAIK.